src/org/opensolaris/opengrok/history/HistoryGuru.java

```diff
@@ -629,9 +629,7 @@ public void createCache(Collection<String> repositories) {
     public List<Repository> clearCache(Collection<String> repositories) throws HistoryException {
         List<Repository> repos = getReposFromString(repositories);
         HistoryCache cache = historyCache;
-        if (cache == null) {
```

Comment:
in the constructor it is in an statement `if (env.useHistoryCache()) {`, is it ensured that this `clearCache` is not called when this is not in effect?